joomla/joomla-cms
# [plg_content_pagebreak] Let users template override TOC

See discussion here: https://github.com/joomla/joomla-cms/pull/17670

### Summary of Changes
- Added possibility to use a template override for TOC.
- Don't break B\C.
- BTW: Fix older bug: "Show All" never has `active` class even if active.

### Testing Instructions
- Plugin "Content - Page Break". Make settings:

![19-04-_2018_19-35-49](https://user-images.githubusercontent.com/20780646/39008439-f5028d6c-4408-11e8-9bda-eb3771ac4598.jpg)

- Add some pagebreak markers to an article and open article in frontend.
- Inspect TOC (also source code).

![19-04-_2018_19-37-41](https://user-images.githubusercontent.com/20780646/39008542-46fd6506-4409-11e8-925e-37bd64bf8c86.jpg)

- Apply patch.
- Inspect TOC (also source code).

- Expected: No changes.

- Create a template override for new file tmpl/toc.php.
Example with Protostar:

![19-04-_2018_19-43-30](https://user-images.githubusercontent.com/20780646/39008786-08df79ac-440a-11e8-81cb-da80d55fb091.jpg)

- Check that the override works.
